src/django/api/limitation/date/date_limitation_context.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+
+from api.limitation.date.date_limitation import (
+    DateLimitation
+)
+from api.limitation.date.blank_date_limitation import (
+    BlankDateLimitation
+)
+from api.limitation.date.monthly_date_limitation import (
+    MonthlyDateLimitation
+)
+from api.limitation.date.yearly_date_limitation import (
+    YearlyDateLimitation
+)
+
+
+class DateLimitationContext:
+    __dateLimitation: DateLimitation
+
+    def __init__(self, renewal_period, start_date):
+        date = self.__prepare_start_date(start_date)
+        self.__dateLimitation = BlankDateLimitation(date)
+
+        if renewal_period == 'MONTHLY':
```

Comment:
Based on the Strategy pattern being used here, the conditionals should be relocated to the client code. The Context class should only be aware of the DateLimitation interface. This approach will help to maintain the Open/Closed principle.

Reply:
Here context is mixed with the factory. I'd rather speak about replacing context with factory method https://refactoring.guru/design-patterns/factory-method and removing context at all.

Reply:
I think this task is more about using different strategies but not about creating different objects. Anyway, there will be a need to use conditionals in the client code.

Reply:
I will take a look to it.